database/migrations/2022_11_06_022852_require_user_group_events_details_and_type.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+use App\Models\UserGroupEvent;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class RequireUserGroupEventsDetailsAndType extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        // The format of `details` before this was not finalised, and on
+        // production nothing important was logged anyway. No going back!
```

Comment:
can someone check if this is true :eyes: 
basically I am expecting `select count(*) from user_group_events where group_id <> 2 or type <> 'user_add'` to be 0.